src/components/common/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+  Button,
+  ButtonGroup,
+  Icon,
+  UseRadioGroupProps,
+  useRadio,
+  useRadioGroup,
+} from "@chakra-ui/react"
+
+interface RadioButtonProps {
+  label?: string
+  icon?: React.ElementType
+  isChecked?: boolean
+  value: string
+}
+
+interface RadioButtonGroupProps {
+  radioGroupProps: UseRadioGroupProps
+  options: Array<{ value: string; label?: string; icon?: React.ElementType }>
+  renderOption?: (option: RadioButtonProps, radioProps: any) => JSX.Element
+  buttonGroupStyleProps?: any
+}
+
+const RadioButtonGroup = (props: RadioButtonGroupProps) => {
+  const { getRootProps, getRadioProps } = useRadioGroup(props.radioGroupProps)
+
+  const group = getRootProps()
+
+  const renderButton = (option, radioProps) => {
+    if (props.renderOption) return props.renderOption(option, radioProps)
+    return <RadioButton {...option} {...radioProps}></RadioButton>
+  }
+
+  return (
+    <ButtonGroup {...props.buttonGroupStyleProps} {...group}>
+      {props.options.map((option) => {
+        const radioProps = getRadioProps({ value: option.value })
+        return renderButton(option, radioProps)
+      })}
+    </ButtonGroup>
+  )
+}
+
+export const RadioButton = (props) => {
```

Comment:
We could also type these props here, I believe it's pretty easy using the types from the top of this file